Remotion/ObjectBinding/Web/Services/BusinessObjectWithIdentityProxy.cs

```diff
@@ -46,18 +47,21 @@ public BusinessObjectWithIdentityProxy (IBusinessObjectWithIdentity obj)
       _displayName = obj.GetAccessibleDisplayName();
     }
 
-    public string UniqueIdentifier
+    [DisallowNull]
+    public string? UniqueIdentifier
     {
       get { return _uniqueIdentifier; }
       set { _uniqueIdentifier = ArgumentUtility.CheckNotNullOrEmpty ("value", value); }
     }
 
-    public string DisplayName
+    [DisallowNull]
+    public string? DisplayName
     {
       get { return _displayName; }
       set { _displayName = ArgumentUtility.CheckNotNullOrEmpty ("value", value); }
     }
 
+    [AllowNull]
```

Comment:
What guard? I added the attribute because the setter handles `null` values, even if its not needed. But it would make sense to drop it, together with the guard + add an argument check.

Reply:
Sorry, I meant "fallback" not guard, my bad. My reasoning was, that there's no need to have the AllowNull since there's no observable reason why someone needs to assign null here. That's my reasoning but it may be flawed :)